mat/hogsvd.go

```diff
@@ -101,7 +101,18 @@ func (gsvd *HOGSVD) Factorize(m ...Matrix) (ok bool) {
 		gsvd.err = errors.New("hogsvd: eigen decomposition failed")
 		return false
 	}
-	v := eig.Vectors()
+	vc := eig.VectorsTo(nil)
+	// vc is guaranteed to have real eigenvalues.
```

Comment:
Do we want to have API for Eigen that will make available real eigenvalues if they exist?

Reply:
Not in this PR.

More generally I'm not sure. We should eventually have a `CDense -> Dense` helper routine. I don't know how common it is to know you have real eigenvalues unless your matrix is symmetric (at which point you should use EigenSym). At least in my experience with Matlab it's quite common that matrices that are supposed to be real are complex with some tolerance, and I'm not sure what to do about those cases.

Reply:
I'll file an issue to consider this.